command_line/fp3.py

```diff
@@ -0,0 +1,578 @@
+import procrunner
+import sys
+import os
+import glob
+import copy
+import concurrent.futures
+import logging
+import time
+
+logger = logging.getLogger("fp3")
+
+from iotbx import phil
+from dxtbx.model.experiment_list import ExperimentList, ExperimentListFactory
+from dials_scratch.fp3 import even_blocks, index_blocks, format_phil_include
+from dials_scratch.fp3 import nproc, combine_reflections, combine_experiments
+from dials_scratch.fp3 import find_setup_script
+
+scope = phil.parse(
+    """
+find_spots {
+  include scope dials.command_line.find_spots.phil_scope
+}
+index {
+  include scope dials.command_line.index.phil_scope
+}
+refine {
+  include scope dials.command_line.refine.phil_scope
+}
+integrate {
+  include scope dials.command_line.integrate.phil_scope
+}
+symmetry {
+  include scope dials.command_line.symmetry.phil_scope
+}
+scale {
+  include scope dials.command_line.scale.phil_scope
+}
+resolution {
+  include scope dials.command_line.estimate_resolution.phil_scope
+}
+parallelism = *process drmaa
+  .type = choice
+max_workers = 1
+  .type = int
+  .help = "Maximum number of concurrent workers"
+worker_nproc = 1
+  .type = int
+  .help = "Number of processes for each worker"
```

Comment:
It would be neat if the default was some kind of appropriate automatic selection for the local machine this is running on

Reply:
Added in b219fc050e14e781f74c76f8b33cdb5f80ac643a

It is possibly a bit of a baroque solution but should do something sensible.